src/Core/NotificationCenter/Commands/UpdateNotificationCommand.cs

```diff
@@ -43,5 +47,7 @@ await _authorizationService.AuthorizeOrThrowAsync(_currentContext.HttpContext.Us
         notification.RevisionDate = DateTime.UtcNow;
 
         await _notificationRepository.ReplaceAsync(notification);
+
+        await _pushNotificationService.PushSyncNotificationUpdateAsync(notification, null);
```

Comment:
:thinking: 
Trying to understand how the client would receive this and determine what to do?

For example:
1. A notification is made for the user
2. The user reads the notification (updates notification status)
3. The notification message is updated
4. This push is triggered, sending back the notification with a null `ReadDate` and `DeletedDate`
5. The client shows the notification as unread now from the push? 
6. Next time the client hits the API it will pick up the notification status as read?

More generally, do we need to determine between a notification update and a notification status update?

Reply:
You are right. Ideally this should be Notification Create and Update and separately NotificationStatusUpdate.
Maybe i change the naming of `NotificationCreate` to `NotificationUpsert` and the `NotificationUpdate` to `NotificationStatusUpsert`.
That way this command can send `NotificationUpsert`, which makes it clear that there have been no update to the status. Thoughts ?

Reply:
What about splitting it into two separate push notification models? 

The original `SyncNotificationPushNotification` without `ReadDate` and `DeletedDate`. 
A new one `SyncNotificationStatusPushNotification` with `ReadDate` and `DeletedDate`.

Then two push types for `SyncNotificationPushNotification`:  `SyncNotificationCreate` and `SyncNotificationUpdate`. This would allow us to know whether the notification is new or not.


And one push type for `SyncNotificationStatusPushNotification` : `SyncNotificationStatusUpdate` which lets us know it's a status update.